charon-ml/src/Substitute.ml

```diff
@@ -49,17 +49,7 @@ let st_substitute_visitor (subst : subst) =
       TArrow (regions, inputs, output)
 
     method! visit_TVar (subst : subst) id = subst.ty_subst id
-
-    method! visit_type_var_id _ _ =
-      (* We should never get here because we reimplemented [visit_TypeVar] *)
-      raise (Failure "Unexpected")
-
     method! visit_CgVar _ id = subst.cg_subst id
-
-    method! visit_const_generic_var_id _ _ =
-      (* We should never get here because we reimplemented [visit_Var] *)
-      raise (Failure "Unexpected")
-
```

Comment:
I had to remove these because `generic_params` stores some `type_var`s which aren't inside `Ty::TVar` and thus they fail here. I'm not sure if we want to substitute on a whole `generic_params` and what the output of that should be. I'm thinking to turn `GenericParams` into a list of an enum, and then we can restrict substitution to only predicates again.

Reply:
I don't get this: why did the behavior change (how come there are type vars which are not insdie `Ty::TVar`?
On the side of Aeneas, I always pay attention to substituting a whole `generic_params` at once. If we get there it is a *bug* (we should put this kind of checks back).

Reply:
There are type vars in `GenericParams.types`

Reply:
And it would make no sense to substitute them, unless we were turning this into a `GenericArgs` (but we'd lose the predicates).

Reply:
I've added those checks back in #271